src/cli/format-results-cache.js

```diff
@@ -0,0 +1,91 @@
+"use strict";
+
+// Inspired by LintResultsCache from ESLint
+// https://github.com/eslint/eslint/blob/c2d0a830754b6099a3325e6d3348c3ba983a677a/lib/cli-engine/lint-result-cache.js
+
+const fileEntryCache = require("file-entry-cache");
+const stringify = require("fast-json-stable-stringify");
+// eslint-disable-next-line no-restricted-modules
+const { version: prettierVersion } = require("../index.js");
+const { createHash } = require("./utils.js");
+
+const optionsHashCache = new WeakMap();
+const nodeVersion = process && process.version;
+
+/**
+ * @param {*} options
+ * @returns {string}
+ */
+function getHashOfOptions(options) {
+  if (optionsHashCache.has(options)) {
+    return optionsHashCache.get(options);
+  }
+  const hash = createHash(
+    `${prettierVersion}_${nodeVersion}_${stringify(options)}`
+  );
+  optionsHashCache.set(options, hash);
+  return hash;
+}
+
+/**
+ * @typedef {{ hashOfOptions?: string }} OurMeta
+ * @typedef {import("file-entry-cache").FileDescriptor} FileDescriptor
+ *
+ * @param {import("file-entry-cache").FileDescriptor} fileDescriptor
+ * @returns {FileDescriptor["meta"] & OurMeta}
+ */
+function getMetadataFromFileDescriptor(fileDescriptor) {
+  return fileDescriptor.meta;
+}
+
+class FormatResultsCache {
+  /**
+   * @param {string} cacheFileLocation The path of cache file location. (default: `.prettiercache`)
+   */
+  constructor(cacheFileLocation) {
+    this.cacheFileLocation = cacheFileLocation;
+    this.fileEntryCache = fileEntryCache.create(
+      /* cacheId */ cacheFileLocation
+    );
+  }
+
+  /**
+   * @param {string} filePath
+   * @param {any} options
+   */
+  existsAvailableFormatResultsCache(filePath, options) {
+    const fileDescriptor = this.fileEntryCache.getFileDescriptor(filePath);
```

Comment:
Feel not safe without file content, but that will be slow.

Reply:
For minimum speed and security, why not check the length of the file content rather than the content of the file?

Reply:
In this case it's not usable in CI for example. ESLint, for example, has [--cache-strategy](https://eslint.org/docs/user-guide/command-line-interface#--cache-strategy) which allows to use metadata or content for detecting changed files

Reply:
Thanks, I've added `--cache-strategy` option!